larq/callbacks.py

```diff
@@ -51,6 +62,30 @@ def set_model(self, model: keras.models.Model) -> None:
                 f'Optimizer must have a "{self.hyperparameter}" attribute.'
             )
 
+    def set_hyperparameter(self, t: int) -> Any:
+        hp = getattr(self.optimizer, self.hyperparameter)
+        try:  # new API
+            hyperparameter_val = keras.backend.get_value(hp)
+            hyperparameter_val = self.schedule(t, hyperparameter_val)
+        except TypeError:  # Support for old API for backward compatibility
+            hyperparameter_val = self.schedule(t)
+        keras.backend.set_value(hp, hyperparameter_val)
+        return hp
+
+    def on_batch_begin(
+        self, batch: int, logs: Optional[MutableMapping[str, Any]] = None
+    ) -> None:
+        if self.update_freq == "step":
+            # We use optimizer.iterations (i.e. global step), since batch only
+            # reflects the batch index in the current epoch.
+            hp = self.set_hyperparameter(self.optimizer.iterations.numpy())
```

Comment:
This won't work for older versions of TensorFlow or when running in graph mode. I think the following might work:
```suggestion
            hp = self.set_hyperparameter(keras.backend.get_value(self.optimizer.iterations))
```
Or we can move the `keras.backend.get_value` inside `set_hyperparameter`

Reply:
This should work, thanks :+1: